### ISSUE TITLE
Slack fails to build on Big Sur

### ISSUE THREAD

**Issue by voanhduy1512** [NixOS/nixpkgs#123440]
**Describe the bug**
I try to install Slack on Big Sur, there is no cache for it and it fails to build locally

**To Reproduce**

```
nix-shell -p slack
```

or

```
git clone https://github.com/NixOS/nixpkgs.git
cd nixpkgs
nix-build -A slack
```

**Expected behavior**
Slack should be built succesfully

**Screenshots**
```
[nix-shell:~/Projects/nixpkgs]$ nix-build -A slack
these derivations will be built:
  /nix/store/fmayr6w5vi5hagr1flsib30n4i6qwnzy-slack-4.15.0.drv
building '/nix/store/fmayr6w5vi5hagr1flsib30n4i6qwnzy-slack-4.15.0.drv'...
unpacking sources
unpacking source archive /nix/store/a09izp47fsdv7ih53ip171yimji3l623-Slack-4.15.0-macOS.dmg
source root is Slack.app
patching sources
configuring
no configure script, doing nothing
building
no Makefile, doing nothing
installing
2021-05-18 02:41:32.375 defaults[22608:468204] Could not write domain com.tinyspeck.slackmacgap; exiting
builder for '/nix/store/fmayr6w5vi5hagr1flsib30n4i6qwnzy-slack-4.15.0.drv' failed with exit code 1
error: build of '/nix/store/fmayr6w5vi5hagr1flsib30n4i6qwnzy-slack-4.15.0.drv' failed
```

**Additional context**
The reason I found is because of 

```
/usr/bin/defaults write com.tinyspeck.slackmacgap SlackNoAutoUpdates -bool YES
```

I can successfully build Slack if remove that line.

**Notify maintainers**
@mmahut 

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```
 - system: `"x86_64-darwin"`
 - host os: `Darwin 20.6.0, macOS 10.16`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.3.11`
 - channels(anhduy): `"darwin"`
 - channels(root): `"darwin, home-manager, nixpkgs-21.11pre319254.b5182c214fa"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: slack
# a list of nixos modules affected by the problem
module:
```

**Comment by voanhduy1512** [comment]
Right, I think `multi-user` makes the different here.

**Comment by JesusMtnez** [comment]
@voanhduy1512 were you able to solve the issue? I am facing the same problem.

### nix-info
```
 - system: `"aarch64-darwin"`
 - host os: `Darwin 21.2.0, macOS 12.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.5.1`
 - channels(root): `"nixpkgs-22.05pre339466.2beba9a23a8"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

**Comment by wegank** [comment]
Closed by #125051 and #193720.
